test/functional/tools/output_filter.xml

```diff
@@ -37,134 +36,113 @@ echo 'test' > p2.reverse
         </collection>
     </outputs>
     <tests>
-        <test expect_num_outputs="3">
+        <test expect_num_outputs="2">
             <param name="produce_out_1" value="true" />
             <param name="filter_text_1" value="foo" />
             <output name="out_1">
                 <assert_contents>
-                    <has_line line="test" />
+                    <has_line line="1" />
                 </assert_contents>
             </output>
             <output name="out_2">
                 <assert_contents>
-                    <has_line line="test" />
-                </assert_contents>
-            </output>
-            <output name="out_3">
-                <assert_contents>
-                    <has_line line="test" />
+                    <has_line line="2" />
                 </assert_contents>
             </output>
         </test>
-        <test expect_num_outputs="2">
+        <test expect_num_outputs="1">
             <param name="produce_out_1" value="true" />
             <param name="filter_text_1" value="bar" /> <!-- fails second filter in out2 -->
             <output name="out_1">
                 <assert_contents>
-                    <has_line line="test" />
-                </assert_contents>
-            </output>
-            <output name="out_3">
-                <assert_contents>
-                    <has_line line="test" />
+                    <has_line line="1" />
                 </assert_contents>
             </output>
         </test>
-        <test expect_num_outputs="1">
+        <!-- tool runs with no outputs should fail -->
```

Comment:
I don't see why that should be the case ?

Reply:
This was not precise enough. Maybe the comment should say: 

```suggestion
        <!-- a test that does not produce any outputs violates the silent assumption that a tool run should have an output, hence expect_test_failure -->
```

Background if planemo finds a test that does not produce outputs it will skip all output checks and just log the assertion to the terminal. So the user will very likely miss this (in particular in CI). 

We might also remove the assertion completely.

Are there tools that do not produce outputs?

Reply:
So .. the tool itself does not fail (I used `expect_test_failure` and not `expect_failure`).

Reply:
> a test that does not produce any outputs violates the silent assumption that a tool run should have an output, hence expect_test_failure

what is this silent assumption ? an export tool doesn't need to create outputs

Reply:
Maybe implicit assumption is better. Anyway the export tool is a good example that the current change is not good.

The bottom line is the following. Currently, no check of the outputs is done if a tool does not produce any output but an assertion is raised. This assertion only appears in the logs and (for instance) planemo users don't see it if they are not paying attention to the logs (which probably only a few people do).

In other words, tool tests are successful if the tool test does not produce any output regardless of if there are any test output assumptions.

Or even shorter: the `AssertionError` short circuits all test outputs assumptions. 

... Maybe just remove the assertion?